### DISCUSSION TITLE
Security Concerns - CleartextStorage ofSensitive Information OWASP

### DISCUSSION THREAD

**Discussion by mmuhamad12** [ionic-team/capacitor#5852] [General]
Some security concerns on some files from capacitor. below are details. anyone can help address this concerns?

Tool used: MOBSF

Standards

CWE: CWE-312: CleartextStorage ofSensitive Information OWASP Top 10: M9: ReverseEngineering OWASP MASVS: MSTG-STORAGE-14

Files

`com/getcapacitor/Bridge.java `
`com/getcapacitor/AppUUID.java`
`com/getcapacitor/Plugin.java`

package.json

`"@capacitor/android": "^3.7.0"`
`"@capacitor/app": "^1.1.1"`
`"@capacitor/browser": "^1.0.7"`
`"@capacitor/camera": "^1.3.1"`
`"@capacitor/core": "^3.7.0"`
`"@capacitor/ios": "^3.7.0"`

**Comment by mlynch** [comment]
For security sensitive applications and storing sensitive data encrypted on the device, you can use our Identity Vault product. We're also happy to discuss more involved security-related questions. The defaults in Capacitor are secure and meant to be unopinionated. https://ionic.io/products/identity-vault
